pkg/client/client.go

```diff
@@ -0,0 +1,53 @@
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ObjectKey identifies a Kubernetes Object.
+type ObjectKey = types.NamespacedName
+
+// Client defines CRUD operations for runtime object in the cluster
+type Client interface {
+	Create(ctx context.Context, obj runtime.Object) error
+	Get(ctx context.Context, key ObjectKey, objExisting runtime.Object) error
+	Update(ctx context.Context, obj runtime.Object) error
+}
+
+// crud is an implementation of the Client interface
+type kubeClient struct {
+	client client.Client
+}
+
+// NewClient returns a crud that can perform
+// CRUD operations on a runtime object
+func NewClient(client client.Client) Client {
+	return &kubeClient{
+		client: client,
+	}
+}
+
+// Create creates a new runtime object
+func (h *kubeClient) Create(ctx context.Context, obj runtime.Object) error {
+
```

Comment:
Please don't add unnecessary lines like this. Comment applies everywhere you are doing this.